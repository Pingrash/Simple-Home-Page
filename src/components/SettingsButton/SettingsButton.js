import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  button: {
    alignSelf: 'flex-end',
    marginTop: '1rem',
    width: 90,
    backgroundColor: '#B3E5FC',
    color: 'black'
  }
}));

const SettingsButton = ({ toggleSettings }) => {
  const classes = useStyles();

  return (
    <Button
      data-test='button'
      className={classes.button}
      onClick={() => toggleSettings()}
      variant='contained'
    >
      Settings
    </Button>
  );
};

SettingsButton.propTypes = {
  toggleSettings: PropTypes.func.isRequired
};

export default SettingsButton;
