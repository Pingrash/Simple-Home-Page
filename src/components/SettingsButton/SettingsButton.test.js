import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {
  findByDataTest,
  checkProps,
  testFunc
} from '../../../Utilities';
import SettingsButton from './SettingsButton';

const props = { toggleSettings: testFunc };

const setup = () => {
  const component = shallow(<SettingsButton {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(<SettingsButton {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SettingsButton renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render a button', () => {
    const button = findByDataTest(wrapper, 'button');
    expect(button.length).toBe(1);
  });
});

describe('SettingsButton should have correct props', () => {
  it('Should not return a props error', () => {
    const propsError = checkProps(SettingsButton, props);
    expect(propsError).toBeUndefined();
  });
});
