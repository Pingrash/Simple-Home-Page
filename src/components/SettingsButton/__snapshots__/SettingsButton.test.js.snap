// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1"
  data-test="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Settings
  </span>
</button>
`;
